grammar.ebnf

```diff
@@ -1,8 +1,15 @@
+(*
+* Copyright: Copyright Ioannis Cheilaris 2019.
+* License:   $(LINK2 http://www.boost.org/LICENSE_1_0.txt, Boost License 1.0)
+* Authors:   Ioannis Cheilaris, Stefanos Baziotis
+*)
+
+
 Module = {Declaration}, ? EOI ? ;
 
 Declaration = StructDecl | FunctionDef;
 
-FunctionDef = "func", Identifier, "(", FuncDefParamList, ")", "{", {Statement},
+FunctionDef = "func", Identifier, "(", FuncDefParamList, ")", [":", Type] "{", {Statement},
```

Comment:
Ok, at the start of the function we should have all variable declarations. So, you can make a `VarDeclaration` rule. We can do it by looking the samples. Don't think of it too hard as it's quite possible that we will have to change it. :)

Reply:
Wouldn't it be better if we could declare variables wherever and not just at the start?

Reply:
It depends on what you mean "better". For the programmer writing in DIL, probably yes (although there are a lot of people that prefer the ANSI C style of declarations - i.e. at the start of the function). But for the programmer implementing DIL (i.e. us), no. It's not such a big of a deal, but better leave unneeded complications for now.